fiftyone/utils/labels.py

```diff
@@ -673,3 +674,116 @@ def classifications_to_detections(
+    helps reduce false positives.
+
+    Args:
+        sample_collection: a
+            :class:`fiftyone.core.collections.SampleCollection`
+        in_field: the name of the :class:`fiftyone.core.labels.Detections` field
+        out_field: the name of the :class:`fiftyone.core.labels.Detections`
+            field to populate. Creates a new field if one doesn't already exist
+            with this name.
+        conf_threshold (0.6): a floating-point value between 0 and 1 representing
+            the minimum confidence score required for a detection to be considered
+            valid. Detections with confidence scores lower than this threshold
+            will be discarded during the Non-Maximum Suppression (NMS) process.
+        iou_threshold (0.5): a floating-point value between 0 and 1 representing
+            the Intersection over Union (IoU) threshold used in the NMS algorithm.
+            It determines the minimum overlap required between bounding boxes for
+            them to be considered duplicates. Bounding boxes with IoU values
+            greater than or equal to this threshold will be suppressed.
+        progress (None): whether to render a progress bar (True/False), use the
+            default value ``fiftyone.config.show_progress_bars`` (None), or a
+            progress callback function to invoke instead
+    """
+    fov.validate_collection_label_fields(
+        sample_collection, in_field, fol.Detections
+    )
+
+    samples = sample_collection.select_fields(in_field)
+
+    with fou.ProgressBar(progress=progress) as pb:
+        for sample in pb(samples):
+            detections_data_json_str = sample[in_field].to_json()
+            nms_detections_data_json_str = _perform_nms(
+                detections_data_json_str, conf_threshold, iou_threshold
+            )
+            nms_processed_detections = fol.Detections.from_json(
+                nms_detections_data_json_str
+            )
+            sample[out_field] = nms_processed_detections
+            sample.save()
+
+
+def _perform_nms(
+    detections_data_json_str, conf_threshold=0.6, iou_threshold=0.5
+):
+    detections_data_json = etas.load_json(detections_data_json_str)
+    detections = detections_data_json["detections"]
+
+    # Sort detections by confidence in descending order
+    detections.sort(key=lambda x: x["confidence"], reverse=True)
+
+    # Remove detections with confidence less than the conf_threshold
+    detections = [d for d in detections if d["confidence"] >= conf_threshold]
+
+    nms_detections = []
+
+    while len(detections) > 0:
+        # Pick the detection with highest confidence
+        selected_detection = detections[0]
+        nms_detections.append(selected_detection)
+        del detections[0]
+
+        # Compare with other detections for NMS
+        for d in detections:
+            if d["label"] == selected_detection["label"]:
+                iou = _calculate_iou(
```

Comment:
Can you use the existing `fiftyone.utils.iou.compute_ious()` utility here to avoid re-implementing this calculation?

Reply:
Sure, I'll incorporate this. It's much better always to use an existing utility! 😄